generator/src/main/java/org/corfudb/generator/Generator.java

```diff
@@ -19,55 +19,201 @@
+     * sends the correctness log (of one chunk) to verification, then resumes
+     * the run.
+     * The same action happens in a cyclic fashion until the totalRuntime.
+     */
+    public void runLongevityTestForVerification() {
+
+        fillOperationQueue();
+        long lastPollTime = System.currentTimeMillis();
+        consumeOperationQueue();
+        runCpTrimOperation();
+
+        while (true) {
+            // One round of 'chunk' starts
+            long cumulativeRuntime = 0;
+            while (true) {
+                Sleep.sleepUninterruptibly(POLL_TIME);
+
+                // Check if the operationQueue is drained or not,
+                // if it's empty and has not reached CHUNK_RUNTIME,
+                // then fill up the queue for further consuming.
+                if (operationQueue.isEmpty()) {
+                    // Calculate the 'cumulative' runtime so far,
+                    // skipping the time for filling up the operation queue
+                    cumulativeRuntime += System.currentTimeMillis() - lastPollTime;
+                    // Break this while loop if it reaches CHUNK_RUNTIME
+                    if (cumulativeRuntime >= CHUNK_RUNTIME.toMillis()) {
+                        // At the end of the chunk, give some margin
+                        // for the workers to finish their operations
+                        Sleep.sleepUninterruptibly(CHUNK_TIMEOUT);
+                        Correctness.recordOperation("Liveness, " + "Success", false);
+                        break;
+                    }
+                    fillOperationQueue();
+                    lastPollTime = System.currentTimeMillis();
+                } else {
+                    // If it's been more than LIVENESS_TIMEOUT, and the queue is still not drained,
+                    // then the system might have got stuck and we record its 'Liveness' as 'Fail'.
+                    if (System.currentTimeMillis() - lastPollTime > LIVENESS_TIMEOUT.toMillis()) {
+                        Correctness.recordOperation("Liveness, " + "Fail", false);
+                    }
                 }
             }
-        };
 
-        Future[] appsFutures = new Future[numThreads];
+            // TODO: send the correctness log (of this chunk) to verification and clean up.
 
-        for (int x = 0; x < numThreads; x++) {
-            appsFutures[x] = appWorkers.submit(app);
-        }
-
-        for (int x = 0; x < numThreads; x++) {
-            try {
-                appsFutures[x].get();
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+            // Minus one CHUNK_RUNTIME from the TOTAL_RUNTIME
+            totalRuntime = totalRuntime.minus(CHUNK_RUNTIME);
+            if (totalRuntime.toHours() == 0) {
+                break;
             }
+            // Fill up the queue for next round of 'chunk' run
+            fillOperationQueue();
+            lastPollTime = System.currentTimeMillis();
```

Comment:
Anny, today I discussed your doubt with Maithem and added a 'TODO: checkpoint all the existing maps of last chunk and do prefix trim' after shipping one chunk of log to verification. Our idea is that instead of recording the last chunk's state on the verification side, we 'patched' the last state to the head of next chunk, i.e. before running the next chunk, we checkpoint all the existing maps of last chunk and do prefix trim, in this case, when next chunk starts running, it'll also have the old data.